edgelet/edgelet-core/src/certificate_properties.rs

```diff
@@ -89,6 +91,15 @@ impl CertificateProperties {
         self.issuer = issuer;
         self
     }
+
+    pub fn san_entries(&self) -> Option<&Vec<String>> {
+        self.san_entries.as_ref()
+    }
+
+    pub fn with_san_entries(mut self, entries: &Vec<String>) -> CertificateProperties {
+        self.san_entries = Some(entries.clone());
```

Comment:
I think we should consume the `entries` parameter and force the caller to clone if they need to.

Reply:
Or alternately, `&[&str]` maybe?